src/middleware.rs

```diff
@@ -159,6 +159,42 @@ impl AfterMiddleware for HandlebarsEngine {
             }
         }
     }
+
+    fn catch(&self, _: &mut Request, e: IronError) -> IronResult<Response> {
```

Comment:
Can't you just do:

    err.response = try!(self.after(request, err.response));
    Err(err)